libmamba/include/mamba/core/channel.hpp

```diff
@@ -26,24 +26,36 @@ namespace mamba
     namespace specs
     {
         class ChannelSpec;
+        class AuthenticationDataBase;
     }
 
-    std::vector<std::string> get_known_platforms();
-
     // Note: Channels can only be created using ChannelContext.
     class Channel
     {
     public:
 
-        Channel(const Channel&) = delete;
-        Channel& operator=(const Channel&) = delete;
-        Channel(Channel&&) noexcept = default;
-        Channel& operator=(Channel&&) noexcept = default;
+        struct ResolveParams
+        {
+            using platform_list = util::flat_set<std::string>;
+            using channel_list = std::vector<Channel>;
+            using channel_map = std::map<std::string, Channel>;
+            using multichannel_map = std::unordered_map<std::string, channel_list>;
+
+            const platform_list& platforms;
+            const specs::CondaURL& channel_alias;
+            const channel_map& custom_channels;
+            const specs::AuthenticationDataBase& auth_db;
+
+            // TODO add CWD and home
+        };
 
-        ~Channel();
 
-        const std::string& location() const;
-        const std::string& name() const;
+        Channel(
+            specs::CondaURL url,
+            std::string canonical_name,
+            util::flat_set<std::string> platforms = {}
+        );
```

Comment:
Now that this constructor is public, do we still need the `make_channel` method in the ChannelContext?